### ISSUE TITLE
Don't clear big render target if all pixels will be replaced

### ISSUE THREAD

**Issue by cosmic-linden** [secondlife/viewer#2638]
### Description

`LLRenderTarget::clear` is called even when the cleared pixels would all be overwritten. Disable where possible for cases where rendering via `mScreenTriangleVB`, as screen-sized render buffers can be big.

## Test plan

There is chance of sudden FPS *decrease* on some hardware. If setting the new debug setting `RenderBufferClearOnInvalidate` to true *increases* the FPS significantly, please fail this test plan.

Keep an eye out for unexpected graphics changes. These changes mainly affect graphics post-processing, including glow, lighting, antialiasing, and screen space reflections (note: screen space reflections are disabled by default). If setting the new debug setting `RenderBufferClearOnInvalidate` to true fixes a graphics regression, please fail this test plan.

**Comment by RyeMutt** [comment]
These changes caused some rendering issues due to breaking render targets that expected to be cleared before use.

**Comment by cosmic-linden** [comment]
Yes @RyeMutt  , iirc you mentioned that removing some of the new clear calls helped prevent annoying flickering. Are we missing the fix for that?

**Comment by RyeMutt** [comment]
I never finished working on that. I was just noting that this set of changes was a source of rendering artifacts if they were to be revived from ```archive/develop```. 

From a quick glance it's likely the calls that removed the explicit clears to 1 could be suspect. Also the change that remove the clear call around the comment ```// clear color buffer here - zeroing alpha (glow) is important or it will accumulate against sky```
